fabric-mining-level-api-v1/src/main/java/net/fabricmc/fabric/api/mininglevel/v1/FabricTool.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.mininglevel.v1;
+
+import org.jetbrains.annotations.Nullable;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.item.ToolMaterial;
+import net.minecraft.tag.Tag;
+
+/**
+ * Interface for tools to provide information about their mining ability.
+ * Allows items that don't extend {@code MiningToolItem}, such as Swords and Shears, to provide mining information.
+ */
+public interface FabricTool {
+	/**
+	 * Returns the {@code ToolMaterial} of this tool.
+	 * Can be {@code null} if the tool does not utilise {@code ToolMaterial}s.
+	 * Defaults to {@code null}.
+	 * @return The tool material
+	 */
+	@Nullable
+	default ToolMaterial getToolMaterial() {
+		return null;
+	}
+
+	/**
+	 * Returns a {@code Tag} of the blocks this tool is effective against.
+	 * Can be {@code null} if the tool does not utilise a tag.
+	 * Defaults to {@code null}.
+	 * @return The effective blocks
+	 */
+	@Nullable
+	default Tag<Block> getEffectiveBlocks() {
+		return null;
+	}
+
+	/**
+	 * Returns the mining level of this tool.
+	 * Defaults to the mining level of the material returned by {@link #getToolMaterial()}, or 0 if that returns {@code null}.
+	 * @return The mining level
+	 */
+	default int getMiningLevel() {
+		ToolMaterial toolMaterial = getToolMaterial();
+
+		if (toolMaterial != null) {
+			return getToolMaterial().getMiningLevel();
```

Comment:
Why not just use the local variable `toolMaterial` here...

Reply:
lol must have missed that when refactoring some things, fixed